### PR TITLE
Broker ingress and Dispatcher now uses the protocol binding instead of json to process events

### PR DESCRIPTION
**Pull request by gabo1208**
Merge after #750 

# Changes

- 🎁 Now the Broker's ingress uses the Binary representation of the CloudEvents
- 🎁 Now the Broker's Dispatcher uses the protocol binding to get a CloudEvent from a RabbitMQ Message
- 🎁 Added performance tests for the source using the new Ingress CE Binary represenation, that makes the Ingress plugable to the Source
- 🧹 Updated Broker's performance tests results to reflect the performance gains (like 10% less latency and less cpu and memory consumption overall)

/kind enhancement performance

Fixes #

**Release Note**

```release-note
- 🎁 Now the Broker's ingress uses the Binary representation of the CloudEvents
- 🎁 Now the Broker's Dispatcher uses the protocol binding to get a CloudEvent from a RabbitMQ Message
- 🎁 Added performance tests for the source using the new Ingress CE Binary represenation, that makes the Ingress plugable to the Source
```

### CODE REVIEW

File: pkg/dispatcher/dispatcher_test.go [hunk 1]
```diff
@@ -158,7 +58,55 @@ func TestFailToConsume(t *testing.T) {
...
+	if err != nil {
+		ch.Close()
+		fakeServer.Stop()
+		return nil, nil, fmt.Errorf("failed to bind Queue: %s", err)
+	}
+	return ch, fakeServer, nil
+}
+
+/* func TestEndToEnd(t *testing.T) {
+	t.Skip()
```
Comment (gab-satchi) [inline pkg/dispatcher/dispatcher_test.go:109]: can we uncomment the block and leave the t.Skip()? And maybe add a dedicated comment why we're briefly skipping these tests

Comment (gab-satchi) [inline pkg/dispatcher/dispatcher_test.go:109]: can we uncomment the block and leave the t.Skip()? And maybe add a dedicated comment why we're briefly skipping these tests

**Review by gab-satchi — COMMENTED** [review]
thanks for the changes. just some minor things

File: pkg/adapter/adapter_test.go [hunk 2]
```diff
@@ -167,8 +167,7 @@ func TestPostMessage_ServeHTTP(t *testing.T) {
 			if err != nil {
 				t.Errorf("Error unmarshaling wanted request body %s %s", tc.reqBody, err)
 			}
-
-			err = json.Unmarshal([]byte(tc.reqBody), &gotBody)
+			err = json.Unmarshal(m.Body(), &gotBody)
```
[the remark this answers is not in the document]
Comment (gabo1208) [reply pkg/adapter/adapter_test.go:170]: yeah here I should use the generated request body, let me fix this

[the remark this answers is not in the document]
Comment (gabo1208) [reply pkg/adapter/adapter_test.go]: In the case of the headers im just cleaning the "Ce-" prefix, but they come from the httptest request

**Review by gab-satchi — APPROVED** [review]
/lgtm

**Pull request merged**